pkg/sriov/sriov.go

```diff
@@ -394,6 +409,19 @@ func (s *sriovManager) ResetVFConfig(conf *sriovtypes.NetConf) error {
 		}
 	}
 
+	// Restore original spoofchk value if needed
+	// Note: This must happen before restoring Administrative MAC, as some drivers may fail to set admin mac to zero
+	// when spoof check is enabled.
+	if conf.OrigSpoofChk != "" {
+		spoofchk_val := true
+		if conf.OrigSpoofChk == "off" {
+			spoofchk_val = false
+		}
+		if err = s.nLink.LinkSetVfSpoofchk(pfLink, conf.VFID, spoofchk_val); err != nil {
+			return fmt.Errorf("failed to restore spoof checking for vf %d to %s: %v", conf.VFID, conf.OrigSpoofChk, err)
+		}
+	}
```

Comment:
we could add an "else" clause here and set `spoofchk` to false. However, this will not necessarily preserve the original state of the system.